Automattic/jetpack
# Social: Update share status modal to use Dataviews component

Share status modal on mobile looks messed up. See https://github.com/Automattic/jetpack/pull/39051#issuecomment-2315521611

## Proposed changes:

* Make use of [DataViews](https://wordpress.github.io/gutenberg/?path=/docs/dataviews-dataviews--docs) component to render the list of shares
* Thus, fix the mobile view

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the share status feature - `define( 'JETPACK_SOCIAL_HAS_SHARE_STATUS', true );`
* Goto the post editor for a post which has some shares or reshare posts to create a few entries (at least 10)
* Click on "View share history" in the publicize panel
* Confirm that the modal and data renders correctly
* Confirm that the modal content is scrollable when the data is more than what can be displayed in the modal at once
* Confirm that the mobile view isn't squeezed and is rather horizontally scrollable

<table>
    <tbody>
        <tr>
            <th>BEFORE</th>
            <th>AFTER</th>
        </tr>
        <tr>
            <th colspan="2">Destop view</th>
        </tr>
        <tr>
            <td>
<img width="969" alt="Screenshot 2024-09-04 at 4 26 45 PM" src="https://github.com/user-attachments/assets/032d8582-395c-48c2-b2dd-e84b87d20b2c">
</td>
            <td>
<img width="968" alt="Screenshot 2024-09-04 at 4 27 11 PM" src="https://github.com/user-attachments/assets/61922b8f-3395-4505-9182-4743f8dcde6f">
</td>
        </tr>
        <tr>
            <th colspan="2">Mobile view</th>
        </tr>
        <tr>
            <td>
<img width="401" alt="Screenshot 2024-09-04 at 4 35 43 PM" src="https://github.com/user-attachments/assets/b8b36c48-f9f8-4d8d-aac2-84010a59214f">
</td>
            <td>
<img width="401" alt="Screenshot 2024-09-04 at 4 35 08 PM" src="https://github.com/user-attachments/assets/07b84638-103f-4395-9bf1-067819fd5f51">
</td>
        </tr>
    </tbody>
</table>
